Core/include/Acts/Seeding/detail/StrawLineFitAuxiliaries.hpp

```diff
@@ -0,0 +1,203 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+#pragma once
+
+#include "Acts/EventData/StationSpacePoint.hpp"
+#include "Acts/Utilities/ArrayHelpers.hpp"
+#include "Acts/Utilities/Logger.hpp"
+#include "Acts/Utilities/detail/Line3DWithPartialDerivatives.hpp"
+
+namespace Acts::Experimental::detail {
+/// @brief Helper class to calculate the residual between a straight line and
+///        a StationSpacePoint measurement as well as the partial derivatives.
+///        The residual is expressed as a 3D vector, where first two components
+///        describe the spatial residual w.r.t. the precision & non-precision
+///        direction, and the last one expresses the residual between the
+///        parametrized time of arrival of the track and the measurement's
+///        recorded time.
+///
+///        For straw type measurements, the precision residual is calculated as
+///        the difference between the signed line distance between straw wire &
+///        line, and the signed drift radius where the sign simply encodes the
+///        left/right amibiguity. If the measurement additionally carries
+///        information about the passage along the wire, the non-precision
+///        residual is then defined to be the distance along the wire.
+///
+///        For strip type measurements, the residual is the distance in the
+///        strip-readout plane between the point along the line that intersects
+///        the plane and the measurement.
+class StrawLineFitAuxiliaries {
+ public:
+  using Line_t = Acts::detail::Line3DWithPartialDerivatives<double>;
+  using Vector = Line_t::Vector;
+  enum FitParIndices : std::size_t {
```

Comment:
- can this be `enum class`?
- `std::size_t` seems rather big for values up to `5`. can this be replaced by default `int`?